src/policy/compressor/compressorspace.rs

```diff
@@ -38,7 +38,13 @@ impl<VM: VMBinding> SFT for CompressorSpace<VM> {
     }
 
     fn get_forwarded_object(&self, object: ObjectReference) -> Option<ObjectReference> {
-        Some(self.forward(object, false))
+        // Check if forwarding addresses have been calculated before attempting
+        // to forward objects
+        if self.forwarding.has_calculated_forwarding_addresses() {
```

Comment:
This is still not exactly correct. When `forwarding` calculates the offsets but does not yet move the object to the new location, what this function returns is an invalid object reference. Currently no function calls `get_forwarded_object` during that period of time, so it appears to be fine.

Reply:
Why do we care about the actual content when we're forwarding references?